robocluster/serial.py

```diff
@@ -0,0 +1,53 @@
+import serial_asyncio
+import asyncio
+import json
+
+BAUDRATE = 115200
+
+
+class SerialDevice:
```

Comment:
Would this be cleaner if we gave it a similar API to `Socket`?

Reply:
Maybe we want to define some kind of asyncio reader/writer API that we can reference...

Reply:
We could also use the asyncio transports: https://docs.python.org/3.6/library/asyncio-protocol.html
It could make adding new transports such as I2C or CAN bus easy as well.

Reply:
I am hesitant to use the asyncio transports as they need specific implementation in the event loop that we are using. It may be easier to implement some kind of Reader, Writer, and ReaderWriter abstract classes with `read()` and `write()`.